src/main/java/com/demo/controller/DriverController.java

```diff
@@ -0,0 +1,69 @@
+package com.demo.controller;
+
+
+import com.demo.model.Driver;
+import com.demo.model.ParkingMeter;
+import com.demo.service.DriverService;
+import com.demo.service.ParkingEventService;
+import com.demo.service.ParkingMeterService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+public class DriverController {
+
+    @Autowired
+    private DriverService driverService;
+    @Autowired
+    private ParkingMeterService parkingMeterService;
+    @Autowired
+    private ParkingEventService parkingEventService;
+
+
+    @GetMapping("/drivers")
+    public List<Driver> getAllDrivers() {
+        return driverService.findAll();
+    }
+
+    @PostMapping("/drivers")
+    public Driver addDriver(
+            @RequestBody Driver myDriver
+    ) {
+        driverService.addNewDriver(myDriver);
+        return myDriver;
+    }
+
+    @PostMapping("start/{driverId}/{parkingMeterId}")
+    public String startParkingMeter(
+            @PathVariable Long driverId,
+            @PathVariable Long parkingMeterId
+    ) {
+        ParkingMeter parkingMeter = parkingMeterService.findById(parkingMeterId);
+        if (parkingMeter.isOccupied()) {
```

Comment:
czy Twoim zdaniem to sprawdzenie wyczerpuje temat podwójnej rezerwacji miejsca?

Reply:
Moim zdaniem tak. Nie obsługujemy rezerwowania parkingu w przyszłym czasie, więc nie powinien pojawić się problem z podwójną rezerwacją miejsca. Jednak bardzo proszę o dodatkową podpowiedź w tym przypadku, jeżeli się mylę.

Reply:
a co się stanie jeśli dla 2 równoległych requestów sprawdzimy tego if-a, dla obu zwróci false, a potem oba pójdą dalej?

Reply:
i teraz ten if jest pusty w ogóle ;)